### PR TITLE
adds in 3.3.1 Ecto.Type stuff

### PR DESCRIPTION
**Pull request by veverkap**
Also ran mix.format against

### CODE REVIEW

**Comment by iteles** [comment]
@veverkap Thank you for the pull request, welcome to dwyl! 🎉 

Because our modules are used by various folks in production (including us!), we aim to keep our code coverage at 100% as an extra safe guard, which is why merging is blocked at the moment.

If you're up for upping the coverage to 100%, that would be fantastic ☺️

**Comment by veverkap** [comment]
I'd be happy to do that - it may take a few days, but I'll ping on this PR when that's done.

File: lib/aes.ex [hunk 1]
```diff
@@ -35,9 +35,11 @@ defmodule Fields.AES do
     key_id = get_key_id()
     key = get_key(key_id)
     {ciphertext, tag} = :crypto.block_encrypt(:aes_gcm, key, iv, {@aad, to_string(plaintext), 16})
-    key_id_str = String.pad_leading(to_string(key_id), 4, "0") # 1 >> "0001"
+    # 1 >> "0001"
```
Comment (nelsonic) [inline lib/aes.ex]: I really wish `mix format` would leave in-line comments alone ... 
moving them to the next line loses the context. 💭 

not the fault of this PR, just the formatter being unhelpful. 👍

File: mix.exs [hunk 2]
```diff
@@ -31,9 +31,12 @@ defmodule Fields.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:argon2_elixir, "~> 2.1.2"},    # password hashing
-      {:ecto, "~> 3.2.5"},             # ecto types
-      {:html_sanitize_ex, "~> 1.3"}, # strip noise from html field
+      # password hashing
```
Comment (nelsonic) [inline mix.exs]: increases versions of dependencies and formats in-line comments to separate lines. 👍

File: test/email_test.exs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 defmodule Fields.EmailTest do
   use ExUnit.Case
-  alias Fields.{EmailPlaintext, EmailHash, EmailEncrypted}
+  alias Fields.{EmailEncrypted, EmailHash, EmailPlaintext}
```
Comment (nelsonic) [inline test/email_test.exs]: alphabetise aliased functions Sort `ASC`. 👍

**Comment by nelsonic** [comment]
@veverkap the new version on Hex.pm https://hex.pm/packages/fields includes your changes. 👍 
Thanks again for your contribution. 🎉

File: lib/password.ex [hunk 4]
```diff
@@ -27,4 +27,8 @@ defmodule Fields.Password do
   def load(value) do
     {:ok, value}
   end
+
+  def embed_as(_), do: :self
```
Comment (nelsonic) [inline lib/password.ex]: We need to add this code to our encryption example see: https://github.com/dwyl/phoenix-ecto-encryption-example/issues/31

**Pull request merged**